extism/extism-fsnotify
# feat: generalize with ability to dynamically create tree of apps

To test:
- checkout `generalize` 
- make -B plugins
- make run

In your Finder or file navigator, open the "apps" directory, and see `invert` and `md2html`. Copy the README or another .md file into `md2html` and should see the program crash with an `index out of range`. 

@zshipko - I can't tell if this is a Go host issue or something in the runtime. My logs show nothing really problematic, so I'm guessing Go - but it's a little deeper than I can go at the moment. If you have a minute to try this out, you might have better insight. 

```
extism::sdk INFO 2022-09-04T11:07:35.056136-06:00 (runtime/src/sdk.rs:35) - New plugin added: 0
extism::sdk DEBUG 2022-09-04T11:07:35.056538-06:00 (runtime/src/sdk.rs:107) - Calling function: handle_file in plugin 0
extism::memory DEBUG 2022-09-04T11:07:35.056549-06:00 (runtime/src/memory.rs:136) - Allocating 1567 bytes
extism::sdk INFO 2022-09-04T11:10:02.392332-06:00 (runtime/src/sdk.rs:35) - New plugin added: 0
extism::sdk DEBUG 2022-09-04T11:10:02.392884-06:00 (runtime/src/sdk.rs:107) - Calling function: handle_file in plugin 0
extism::memory DEBUG 2022-09-04T11:10:02.392901-06:00 (runtime/src/memory.rs:136) - Allocating 1579 bytes
extism::memory DEBUG 2022-09-04T11:10:02.393091-06:00 (runtime/src/memory.rs:136) - Allocating 1 bytes
extism::sdk DEBUG 2022-09-04T11:10:02.393111-06:00 (runtime/src/sdk.rs:107) - Calling function: on_file_write in plugin 0
extism::memory DEBUG 2022-09-04T11:10:02.393117-06:00 (runtime/src/memory.rs:136) - Allocating 1579 bytes
extism::memory DEBUG 2022-09-04T11:10:02.393212-06:00 (runtime/src/memory.rs:136) - Allocating 1755 bytes
extism::sdk DEBUG 2022-09-04T11:10:02.393661-06:00 (runtime/src/sdk.rs:107) - Calling function: handle_file in plugin 0
extism::memory DEBUG 2022-09-04T11:10:02.393718-06:00 (runtime/src/memory.rs:136) - Allocating 1738 bytes
extism::memory DEBUG 2022-09-04T11:10:02.393752-06:00 (runtime/src/memory.rs:136) - Allocating 0 bytes

```
